.licenses/go/golang.org/x/crypto/poly1305.dep.yml

```diff
@@ -0,0 +1,63 @@
+---
+name: golang.org/x/crypto/poly1305
```

Comment:
Why is this lib requied?

Reply:
I don't know why it was added, but I see that there is a defect in the "**Check Go Dependencies**" workflow:

https://github.com/arduino/arduino-cli/blob/67924a303cb7ef2473496d1673f10d5412c74967/.github/workflows/check-go-dependencies-task.yml#L67

This will not detect files that were removed by the `general:cache-dep-licenses` task execution of the previous workflow step. It only detects changes to files and added files.

The correct command:
```text
          if ! git diff --color --exit-code HEAD; then
```

Demo: https://github.com/per1234/arduino-cli/runs/6029043113?check_suite_focus=true

Superfluous files in the dependency license metadata cache won't actually interfere with its intended functionality, but the cache is already pretty big so we should try to avoid an accumulation of unused files.

Reply:
Nice discovery Per! Should we open a new PR for this bug fix?

Reply:
If you open a PR to fix the workflow, I'll rebase this one and remove this commit.

Reply:
I'm preparing the PR now.

Reply:
>I'll rebase this one and remove this commit.

OK, it is merged in and ready for rebase.